homeassistant/components/wyoming/stt.py

```diff
@@ -0,0 +1,154 @@
+"""Support for Wyoming speech to text services."""
+from collections.abc import AsyncIterable
+import logging
+
+from wyoming.asr import Transcript
+from wyoming.audio import AudioChunk, AudioStart, AudioStop
+from wyoming.client import AsyncTcpClient
+from wyoming.info import Info
+
+from homeassistant.components import stt
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import DOMAIN, SAMPLE_CHANNELS, SAMPLE_RATE, SAMPLE_WIDTH
+from .error import WyomingError
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Wyoming speech to text."""
+    wyoming_info = hass.data[DOMAIN][config_entry.entry_id]["info"]
+    hass.data[DOMAIN][config_entry.entry_id]["provider"] = WyomingSttProvider(
+        hass,
+        config_entry.data[CONF_HOST],
+        config_entry.data[CONF_PORT],
+        wyoming_info,
+    )
+
+    await async_load_platform(
```

Comment:
Create an entity and add it to `async_add_entities` passed into your function. See for example the VoIP binary sensor